src/semaphore_internal.h

```diff
@@ -29,6 +29,13 @@
 
 struct dispatch_queue_s;
 
+#if USE_FUTEX_SEM
+typedef struct {
+	/* Low 32 bits are the count; high bits are the waiter count. */
+	volatile uint64_t dfx_data;
+} dispatch_futex_t;
+#endif  /* USE_FUTEX_SEM */
```

Comment:
that's not what it should be, you should still use the 64bit field everywhere, only "futex" needs to use `&dfx->dfx_futex` for its argument. you wouldnt use the dfx_waiters part of the struct ever, it's mostly for documentation purposes of how the code works, it's important for audit purposes.